File: mobile/lib/shared/models/pdf.dart
class PdfDocument {
  final String id;
  final String userId;
  final String filename;
  final int pageCount;
  final int fileSize;
  final String? summary;
  final bool isSummarized;
  final String uploadedAt;

  const PdfDocument({
    required this.id,
    required this.userId,
    required this.filename,
    required this.pageCount,
    required this.fileSize,
    this.summary,
    required this.isSummarized,
    required this.uploadedAt,
  });

  factory PdfDocument.fromJson(Map<String, dynamic> json) => PdfDocument(
        id: json['id'] as String,
        userId: json['user_id'] as String,
        filename: json['filename'] as String,
        pageCount: json['page_count'] as int,
        fileSize: json['file_size'] as int,
        summary: json['summary'] as String?,
        isSummarized: json['is_summarized'] as bool,
        uploadedAt: json['uploaded_at'] as String,
      );

  String get fileSizeLabel {
    if (fileSize < 1024) return '$fileSize B';
    if (fileSize < 1024 * 1024) {
      return '${(fileSize / 1024).toStringAsFixed(1)} KB';
    }
    return '${(fileSize / (1024 * 1024)).toStringAsFixed(1)} MB';
  }
}

File: backend/app/models/pdf.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, Integer, DateTime, ForeignKey, Text, Boolean
from sqlalchemy.orm import Mapped, mapped_column, relationship
from app.core.database import Base


class PdfDocument(Base):
    __tablename__ = "pdf_documents"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    user_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    filename: Mapped[str] = mapped_column(String(255), nullable=False)
    storage_path: Mapped[str] = mapped_column(String(500), nullable=False)
    page_count: Mapped[int] = mapped_column(Integer, default=0)
    file_size: Mapped[int] = mapped_column(Integer, default=0)   # bytes
    summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    is_summarized: Mapped[bool] = mapped_column(Boolean, default=False)
    uploaded_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

    owner: Mapped["User"] = relationship("User")  # noqa: F821
